openpype/hosts/maya/plugins/publish/validate_scene_set_workspace.py

```diff
@@ -44,4 +48,35 @@ def process(self, context):
         root_dir = cmds.workspace(query=True, rootDirectory=True)
 
         if not is_subdir(scene_name, root_dir):
-            raise RuntimeError("Maya workspace is not set correctly.")
+
+            if not self.publish_from_published_workfiles:
+                raise RuntimeError("Maya workspace is not set correctly.")
+            else:
+                settings = context.data.get('project_settings')
+                template_name_profiles = settings.get('global') \
+                    .get('publish') \
+                    .get('IntegrateAssetNew') \
+                    .get('template_name_profiles')
+                task_name = context.data["anatomyData"]["task"]["name"]
+                task_type = context.data["anatomyData"]["task"]["type"]
+                key_values = {
+                    "families": "workfile",
+                    "tasks": task_name,
+                    "hosts": context.data["hostName"],
+                    "task_types": task_type
+                }
+                profile = filter_profiles(
+                    template_name_profiles,
+                    key_values,
+                    logger=self.log
+                )
+                anatomy = context.data.get('anatomy')
+                anatomy_filled = anatomy.format(
+                    instance.data.get('anatomyData')
+                )
+                pub_workfile_path = anatomy_filled.get(
+                    profile["template_name"]
+                ).get("folder")
```

Comment:
Better to set an empty string as the default value of key `"folder"`, since the `os.path.realpath()` in `is_subdir()` doesn't work if the input is None type.